app/v1/controllers/validators/CreateAndAmendOtherDeductionsValidatorFactory.scala

```diff
@@ -0,0 +1,109 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.validators
+
+import api.controllers.validators.Validator
+import api.controllers.validators.resolvers.{DateRange, DateRangeResolving, ResolveJsonObject, ResolveNino, ResolveParsedNumber, ResolveTaxYear}
+import api.models.errors._
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import cats.implicits._
+import config.AppConfig
+import play.api.libs.json.JsValue
+import v1.models.request.createAndAmendOtherDeductions.{CreateAndAmendOtherDeductionsBody, CreateAndAmendOtherDeductionsRequestData, Seafarers}
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class CreateAndAmendOtherDeductionsValidatorFactory @Inject() (appConfig: AppConfig) {
+
+  private val customerRefRegex = "^[0-9a-zA-Z{À-˿’}\\- _&`():.'^]{1,90}$".r
+
+  private val resolveJson = new ResolveJsonObject[CreateAndAmendOtherDeductionsBody]()
+
+  def validator(nino: String, taxYear: String, body: JsValue): Validator[CreateAndAmendOtherDeductionsRequestData] =
+    new Validator[CreateAndAmendOtherDeductionsRequestData] {
+
+      def validate: Validated[Seq[MtdError], CreateAndAmendOtherDeductionsRequestData] =
+        (
+          ResolveNino(nino),
+          ResolveTaxYear(appConfig.minimumPermittedTaxYear, taxYear, None, None),
+          resolveJson(body)
+        ).mapN(CreateAndAmendOtherDeductionsRequestData) andThen validateBodyFieldFormat
+
+    }
+
+  private def validateBodyFieldFormat(
+      parsed: CreateAndAmendOtherDeductionsRequestData): Validated[Seq[MtdError], CreateAndAmendOtherDeductionsRequestData] = {
+
+    val validationResult = parsed.body.seafarers match {
+      case None => Valid(())
+      case Some(seafarers) =>
+        seafarers.zipWithIndex
+          .traverse_ { case (item, index) => validateSeafarers(item, index) }
+    }
+
+    validationResult.map(_ => parsed)
+  }
+
+  private def validateSeafarers(seafarers: Seafarers, arrayIndex: Int): Validated[Seq[MtdError], Unit] = {
+    import seafarers._
+
+    (
+      validateCustomerReference(customerReference, s"/seafarers/$arrayIndex/customerReference"),
+      validateAmountDeducted(amountDeducted, s"/seafarers/$arrayIndex/amountDeducted"),
+      validateNameOfShip(nameOfShip, s"/seafarers/$arrayIndex/nameOfShip"),
+      validateDates(fromDate, toDate, arrayIndex)
+    ).tupled
+      .andThen { case (_, _, _, _) => Validated.Valid(()) }
```

Comment:
can just be Valid(())

Reply:
actually the Valid(()) appears 5 times, might be better as a `private val valid = ..`